### PR TITLE
propagate addCommand to children for multiremote

### PR DESCRIPTION
**Pull request by ccharnkij**
## Proposed changes

For #13836, pass the new command down to the children instances of multiremote browser.

## Types of changes

[//]: # (What types of changes does your code introduce to WebdriverIO?)
[//]: # (_Put an `x` in the boxes that apply_)

- [ ] Polish (an improvement for an existing feature)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update (improvements to the project's docs)
- [ ] Specification changes (updates to WebDriver command specifications)
- [ ] Internal updates (everything related to internal scripts, governance documentation and CI files)

## Checklist

[//]: # (_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._)

- [X] I have read the [CONTRIBUTING](https://github.com/webdriverio/webdriverio/blob/main/CONTRIBUTING.md) doc
- [X] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added the necessary documentation (if appropriate)
- [ ] I have added proper type definitions for new commands (if appropriate)

### CODE REVIEW

**Comment by christian-bromann** [comment]
@ccharnkij thanks for contributing to the project.

Something I would like to understand better is how attaching the custom command to every instance works exactly. Note that when you register a custom command to a multiremote browser, the instance you operate within the custom command will act as a multiremote browser, e.g.:

```ts
import { multiremotebrowser } from '@wdio/globals'

multiremotebrowser.addCustomCommand('concatePageTitles', async function (this: WebdriverIO.MultiremoteBrowser) {
	const titles = await this.getTitle() // returns string[]
    return titles.join(' - ')
})

console.log(await multiremotebrowser.concatePageTitles()) // prints "Page A - Page B - Page C
```

Attaching this custom command to an instance wouldn't make sense because `this.getTitle` returns a different type of result.

I haven't really explored this use case thoroughly but this is what I expected it to work. Mind providing more context?

**Comment by ccharnkij** [comment]
Yes, you are correct on that. The ```this``` object for the ```browser``` and instances are different and if, using your example, the call is made from an instance, then the output would only be the title of the page that an instance is on. I think the use case will depend on the user. The user will have take into account the different ```this```, if it's being used, when making a new function.

**Comment by christian-bromann** [comment]
@ccharnkij thanks for clarifying, mind pushing a change to the `Custom Commands` docs on how to handle custom commands with Multiremote?

**Comment by ccharnkij** [comment]
better?

**Comment by ccharnkij** [comment]
Looks like this here is the issue from the latest commits to main in the element.test.ts.
```js
const combinedPrototype = [
        ...new Set([...elementPrototype, ...files])
    ]
```

File: website/docs/CustomCommands.md [hunk 1]
```diff
@@ -97,6 +97,48 @@ Be careful to not overload the `browser` scope with too many custom commands.
...
+*/
+
+browserA.getUrlAndTitle()
+/*
+{
+    url: 'https://webdriver.io/',
+    title: 'WebdriverIO · Next-gen browser and mobile automation test framework for Node.js | WebdriverIO',
+    customVar: undefined
+}
+*/
```
Comment (christian-bromann) [inline website/docs/CustomCommands.md:137]: Maybe we can make this more explicit by using TypeScript:

```ts
import { multiremotebrowser } from '@wdio/globals'

multiremotebrowser.addCommand('getUrlAndTitle', async function (this: WebdriverIO.MultiRemoteBrowser, customVar: any) {
    // `this` refers to:
    //      - MultiRemoteBrowser scope for browser
    //      - Browser scope for instances
    return {
        url: await this.getUrl(),
        title: await this.getTitle(),
        customVar: customVar
    }
})

multiremotebrowser.getUrlAndTitle()
/*
{
    url: [ 'https://webdriver.io/', 'https://webdriver.io/' ],
    title: [
        'WebdriverIO · Next-gen browser and mobile automation test framework for Node.js | WebdriverIO',
        'WebdriverIO · Next-gen browser and mobile automation test framework for Node.js | WebdriverIO'
    ],
    customVar: undefined
}
*/

multiremotebrowser.getInstance('browserA').getUrlAndTitle()
/*
{
    url: 'https://webdriver.io/',
    title: 'WebdriverIO · Next-gen browser and mobile automation test framework for Node.js | WebdriverIO',
    customVar: undefined
}
*/
```

File: website/docs/CustomCommands.md [hunk 2]
```diff
@@ -97,6 +97,48 @@ Be careful to not overload the `browser` scope with too many custom commands.
...
+```
+
+
+
```
Comment (christian-bromann) [inline website/docs/CustomCommands.md:141]:
```suggestion

```

File: website/docs/CustomCommands.md (hunk 1 above)
Comment (ccharnkij) [reply website/docs/CustomCommands.md]: better?

File: website/docs/CustomCommands.md (hunk 2 above)
Comment (ccharnkij) [reply website/docs/CustomCommands.md]: Got it. I squashed and rebased too.

**Review by christian-bromann — APPROVED** [review]
LGTM 👍

**Comment by ccharnkij** [comment]
@christian-bromann Btw, I saw another issue on implementing the UI. Not sure how helpful I can be, but I'd love to join.

**Comment by christian-bromann** [comment]
@ccharnkij awesome, please reach out in the contributor channel on Discord. Thanks!

**Pull request merged**